modules/seqexec/server/src/main/scala/seqexec/server/package.scala

```diff
@@ -125,15 +129,37 @@ package object server {
       ab.fold(a => Validated.Invalid(NonEmptyList.of(a)), b => Validated.Valid(b))
   }
 
-  implicit class ExecutionQueueOps(q: ExecutionQueue) {
-    def status(st: EngineState): SequenceState = {
-      val statuses: List[SequenceState] = q.map(st.executionState.sequences.get(_).map(_.status).getOrElse(SequenceState.Idle))
+  // This assumes that there is only one instance of e in l
+  private def moveElement[T](l: List[T], e: T, d: Int)(implicit eq: Eq[T]): List[T] = {
+    val idx = l.indexOf(e)
 
-      statuses.find(_.isRunning).orElse(statuses.find(_.isError)).orElse(statuses.find(_.isStopped))
-        .orElse(statuses.find(_.isIdle)).getOrElse(SequenceState.Completed)
+    if(d === 0 || idx<0) l
+    else {
+      val (h, t) = l.filterNot(_ === e).splitAt(idx+d)
+      (h :+ e) ++ t
     }
   }
 
+  implicit class ExecutionQueueOps(q: ExecutionQueue) {
```

Comment:
Can it be a value class? `extends AnyVal`?

Reply:
Yes. I'll change it.